src/components/App/App.js

```diff
@@ -43,36 +53,136 @@ function App() {
     return () => document.removeEventListener("keydown", closePopupByOutsideClick);
   }, []);
 
-  function handleSignupClick() {
-    setIsSignupPopupOpen(true);
-  }
+  useEffect(() => {
+    if (token) {
+      mainApi.loadUserInfo(token)
+        .then((res) => {
+          if (res) {
+            setLoggedIn(true);
+            setCurrentUser({ name: res.name, _id: res._id });
+          }
+        })
+        .catch((err) => console.log(err));
+    }
+  }, [token]);
+
+  useEffect(() => {
+    if (token) {
+      mainApi.loadSavedArticles()
+        .then((res) => {
+          setSavedCards(res);
+        })
+        .catch((err) => console.log(err));
+    }
+  }, [token]);
 
   function handleSigninClick() {
     closeAllPopups();
     setIsSigninPopupOpen(true);
   }
 
-  function handleSignin() {
-    //TODO
-  }
-
   function handleMenuClick() {
     setIsMenuOpen(!isMenuOpen);
   }
 
-  function handleSignup() {
-    //TODO
-    setIsInfoPopupOpen(true);
+  function getCardsFromLocalStorage(){
+    const loadedCards = localStorage.getItem("cards");
+    if(loadedCards == null) return [];
+    else return JSON.parse(loadedCards);
+  }
+
+  function handleSignup({ email, password, name }) {
+    console.log("Register : " + email + " , " + password + " , " + name);
+    auth.signup({ email, password, name })
+      .then((res) => {
+        if (res) {
+          setIsSignupPopupOpen(false);
+          setIsInfoPopupOpen(true);
+        }
+      }).catch((err) => {
+        setSubmitError("User with this email address already exists");
+      })
+  }
+
+  function handleSignin({ email, password }) {
+    console.log("Login : " + email + " , " + password);
+    auth.signin({ email, password })
+      .then((res) => {
+        console.log(res);
+        localStorage.setItem("jwt", res.token);
+        setToken(res.token);
+        setLoggedIn(true);
+        setIsSigninPopupOpen(false);
+      })
+      .catch((err) => {
+        setSubmitError("Incorrect email or password");
+      });
+  }
+
+  function handleSignout() {
+    localStorage.removeItem("jwt");
+    setLoggedIn(false);
+    setCurrentUser({ name: "", _id: "" })
+    setToken("");
+    closeAllPopups();
   }
 
   function toggleSigningPopups() {
     setIsSigninPopupOpen(!isSigninPopupOpen);
     setIsSignupPopupOpen(!isSignupPopupOpen);
   }
 
+  function isSavedCard(cardTitle) {
+    return savedCards.length > 0 && savedCards.find(card => card.title === cardTitle);
+  }
+
+  function deleteCard(card) {
+    mainApi.deleteArticle(card._id).then((res) => {
+      setSavedCards((cards) =>
+        cards.filter((item) => item._id !== card._id)
+      );
+    })
```

Comment:
You need to `catch` possible errors at the very end of any server request.

Please, correct it everywhere